covid19dp_submission/steps/download_analysis.py

```diff
@@ -0,0 +1,127 @@
+            analysis_for_processing = analysis_for_processing + unprocessed_analysis[
+                                                                :(num_analysis - len(analysis_for_processing))]
+            break
+        else:
+            analysis_for_processing = analysis_for_processing + unprocessed_analysis
+            offset = offset + limit
+
+    return analysis_for_processing
+
+
+def get_analysis_from_ena(project, offset, limit):
+    analysis_url = f"https://www.ebi.ac.uk/ena/portal/api/filereport?result=analysis&accession={project}&offset={offset}" \
+                   f"&limit={limit}&format=json&fields=run_ref,analysis_accession,submitted_ftp"
+    response = requests.get(analysis_url)
+    if response.status_code != 200:
+        logger.error(f"Error fetching analysis info from ENA for {project}")
+        response.raise_for_status()
+    return response.json()
+
+
+def filter_out_processed_analysis(analysis_array, processed_analysis):
+    unprocessed_analysis = []
+    for analysis in analysis_array:
+        if analysis['analysis_accession'] not in processed_analysis:
+            unprocessed_analysis.append(analysis)
+
+    return unprocessed_analysis
+
+
+def get_processed_analysis(processed_analysis_file):
+    processed_analysis = set()
+    with open(processed_analysis_file, 'r') as file:
+        for line in file:
+            processed_analysis.add(line.split(",")[0])
+    return processed_analysis
+
+
+def download_files(analysis_array, download_target_dir, processed_analysis_file):
+    logger.info(f"total number of files to download: {len(analysis_array)}")
+    with open(processed_analysis_file, 'a') as f:
+        for analysis in analysis_array:
+            download_url = f"http://{analysis['submitted_ftp']}"
+            download_file_name = f"{analysis['analysis_accession']}.vcf"
+            download_file_path = f"{download_target_dir}/{download_file_name}"
+
+            logger.info(f"downloading file {download_url}")
+            urllib.request.urlretrieve(download_url, download_file_path)
+
+            logger.info(f"downloaded file {download_file_name}")
+            f.write(f"{analysis['analysis_accession']},{analysis['submitted_ftp']}\n")
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Download analysis for processing from the Covid-19 DP project',
+                                     formatter_class=argparse.RawTextHelpFormatter, add_help=False)
+    parser.add_argument("--project", default='PRJEB45554', required=False,
+                        help="project from which analysis needs to be downloaded")
+    parser.add_argument("--num-analysis", type=int, default=10000, required=False,
+                        help="Number of analysis to download (max = 10000)")
+    parser.add_argument("--processed-analysis-file",
+                        default="/nfs/production3/eva/data/PRJEB45554/covid19_analysis.txt",
```

Comment:
Suggest not using cluster paths in public repositories..

Reply:
removed default path and made the parameter mandatory